src/app/game/board/board.component.js

```diff
@@ -7,24 +7,42 @@ import DeckService from "./deck/deck.service";
 
 class Board extends Component {
     constructor(props) {
-        super(props)
-        {
-
+        super(props);
+        this.state={
+            discardPile: this.props.discardPile,
+            drawPile: this.props.drawPile,
+            humanHandPile: this.props.humanHand,
+            botHandPile: this.props.botHand
         }
+        this.moveCard = this.moveCard.bind(this);
     }
 
     render() {
         return (
             <div className="board-component">
-                <Hand {...this.props.botHand} />
+                <Hand hand={this.props.botHand}
+                      moveCardDriver1={this.moveCard}
+                />
 
                 <Deck service={DeckService}
                       drawPile={this.props.drawPile}
-                      discardPile={this.props.discardPile} />
-                <Hand {...this.props.humanHand} />
+                      discardPile={this.props.discardPile}
+                      moveCardDriver0={this.moveCard}
+
+                />
+                <Hand hand={this.props.humanHand}
+                      moveCardDriver1={this.moveCard}
+                />
             </div>
         );
     }
+
+    moveCard(card, sourcePile) {
+        this.props.boardService.moveCard(card, this.state[sourcePile.name]);
+        console.log(this.state);
+        this.forceUpdate(); // TODO: ask Offer how to avoid this?
```

Comment:
We should find a better way to do this. This is a bad practice like Offer wrote in his slides

Reply:
Done